server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2730,6 +2730,37 @@ public void updateTabletDataFile(long maxCommittedTime, FileRef newDatafile, Fil
 
   }
 
+  public void updateLastLocation(){
+    synchronized (timeLock) {
+      Tablet t = tabletServer.getOnlineTablet(extent);
+      SortedMap<FileRef, DataFileValue> sm = t.getDatafiles();
+      for (Map.Entry<FileRef, DataFileValue> m : sm.entrySet()) {
```

Comment:
A for loop over the files does not seem right here.  We only need to change the last location of the tablet once.

Reply:
This will be removed since I will not pass in dfv or fileref anymore.